app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100, hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def remove_from_alive(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def check(self) -> None:
+        if self.health <= 0:
+            self.remove_from_alive()
+
+    def __repr__(self) -> str:
+        return f'{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}'
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, other: "Herbivore") -> None:
```

Comment:
The type hint for the `other` parameter in the `bite` method is set to `Herbivore`. This could be restrictive if you later decide to allow `Carnivores` to bite other `Carnivores` or different types of animals. Consider using a more general type hint, like `Animal`, if you plan to extend this functionality.